packages/snaps-controllers/src/snaps/location/npm.ts

```diff
@@ -267,6 +270,13 @@ async function fetchNpmTarball(
   if (!tarballResponse.ok || !tarballResponse.body) {
     throw new Error(`Failed to fetch tarball for package "${packageName}".`);
   }
+  const tarballSizeString = tarballResponse.headers.get('content-length');
```

Comment:
Please add a comment that we're implicitly trusting `npmjs.org` to be a good actor here.

The [content-length can be spoofed](https://stackoverflow.com/a/49933667)

Reply:
@ritave If the `Content-Length` was spoofed, i.e., it is less than the actual size of the tarball, wouldn't that result in just part of the tarball being downloaded?

Reply:
@Mrtenz undefined behaviour and browser specific. IIRC some browsers in the past just kept downloading as long as the stream was open

Reply:
Would depend on how fetch is implemented in the browser I would guess 🤷